common/quotas/global/collection/internal/fallback.go

```diff
@@ -203,9 +203,8 @@ func (b *FallbackLimiter) Wait(ctx context.Context) error {
 }
 
 func (b *FallbackLimiter) Reserve() clock.Reservation {
-	res := b.both().Reserve()
 	return countedReservation{
-		wrapped: res,
+		wrapped: b.both().Reserve(),
 		usage:   &b.usage,
 	}
```

Comment:
main flaw was here: it didn't pass through the Used(false) call when rejected, and did not count rejections.